### PR TITLE
Get latest when package versions are incompatible in merge resolution

### PR DESCRIPTION
**Pull request by pramodya1994**
- Fixes https://github.com/ballerina-platform/ballerina-lang/issues/32860

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by azinneera** [comment]
I think we should do the same at the callee function in DefaultPackageResolver when merging the versions from dist and central.

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/repositories/RemotePackageRepository.java [hunk 1]
```diff
@@ -314,6 +314,14 @@ private PackageNameResolutionRequest toPackageNameResolutionRequest(List<ImportM
 
             if (versionCompatibilityResult.equals(SemanticVersion.VersionCompatibilityResult.GREATER_THAN)) {
```
Comment (azinneera) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/repositories/RemotePackageRepository.java:315]: Do we need to have this old comparison logic too?? Can we just use the one in AbstractPackageRepository??

Comment (hevayo) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/repositories/RemotePackageRepository.java:315]: +1 the idea was to use the new function for comparison

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/repositories/AbstractPackageRepository.java [hunk 2]
```diff
@@ -208,7 +208,7 @@ protected PackageVersion findLatest(List<PackageVersion> packageVersions) {
         return latestVersion;
     }
 
-    protected PackageVersion getLatest(PackageVersion v1, PackageVersion v2) {
+    public PackageVersion getLatest(PackageVersion v1, PackageVersion v2) {
```
Comment (sameerajayasoma) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/repositories/AbstractPackageRepository.java:211]: It is wrong to make this public IMO. We need to move these methods to utility classes. We can do this later anyway!

**Review by sameerajayasoma — APPROVED** [review]

**Review by azinneera — APPROVED** [review]

**Review by azinneera — APPROVED** [review]

**Pull request merged**